Terminal.Gui/Application.cs

```diff
@@ -1432,6 +1435,9 @@ private static void OnUnGrabbedMouse (View view)
 
     // Used by OnMouseEvent to track the last view that was clicked on.
     internal static View? _mouseEnteredView;
+    internal static View? _lastViewButtonPressed;
```

Comment:
We should not be adding more static fields to View. We've learned the hard way that static fields are dangerous and error-prone. We should be moving to REMOVE static fields from View.

I think these pieces of data should be part of `Application` (perhaps encapsulated in a class that `Application` holds a static reference to... e.g. `class MouseState`, or `ConsoleDriver`.

One way to think about this is:

- For instance objects (like `View`) avoid static fields
- `Application` knows what a `View` is and it's ok for `Application` to hold references to `View` objects or `view`-specific objects.
- `ConsoleDriver` should NOT know what `View` is, but since `ConsoleDriver`s are singletons, like `Application`, they are good places to hold persistent data.

Do you agree?

Reply:
I agree for not use them in the View class, but this is in the `Application` which is static. I only made them internal for unit tests purposes and not for use them on the `ConsoleDriver`.

Reply:
I also think singletons for this purpose aren't ideal, and are treating Terminal.Gui like an application more than a framework/library. Every static means something a consumer can now only have one instance of in the _entire_ consuming AppDomain, which is problematic for anyone trying to make a multi-user application.

We do need to remember our place in the stack from the consumer perspective, after all. We don't own the app. It owns Terminal.Gui.

My SnapsInAZfs application that uses Terminal.Gui, for example, doesn't even _call_ anything related to terminal.gui unless specifically told to, for example. And using statics for things means that, if users attach to it at the same time (planned future feature), state is shared and that's going to break a lot of things, with the only way to work around it being to run in different processes and have to use IPC mechanisms.

Reply:
Well... The planned feature is actually aimed more at letting the same user have multiple UIs up that are using the same process, such as in different panes in a tmux session. Different users are a whole different ballgame.

From my perspective as a consumer of the library (any UI library, for that matter), the same process should be able to host as many instances of Terminal.Gui as I want, so long as they each get handed their own screen buffer. But statics that aren't truly just type properties and are instead simulated globals make that dangerous or impossible.

Reply:
And, to be clear, I realize this isn't the only one and there are many others. But they're all resolvable and avoiding adding more will be a big help for that.

Reply:
Making properties and fields instance instead of static is something for another time and discussion, but I thought I'd mention here since it was already called out.

Reply:
I understand that static components may be safer in cases of threading, but in the case of the mouse or keyboard it makes no sense to create separate instances because they are treated as static for the entire application. Therefore, the greatest care to take into account is to ensure that their values correspond to their current states and are not from previous residual values.

Reply:
That's not a safe assumption to make, inside or outside the code of Terminal.Gui.

Any concurrent operation from the same or another thread, even within Terminal.Gui itself, can access anything static like that.

If static, it needs careful attention to synchronization, via locking, semaphores, etc., but that is a non-trivial and easily bug-prone task.

Reply:
I don't think you need to close it, really.

Probably plenty of it that can be used, just not that specific piece.

Or we could close, and if you just don't delete the branch, we can still refer to any parts we need, when we get to it.